package/MDAnalysis/core/universe.py

```diff
@@ -992,6 +993,194 @@ def add_Segment(self, **attrs):
+        types : types (optional, default None)
+            Type of angle, or atom types
+        guessed : bool (optional, default False)
+            Whether these angles have been guessed.
+        order : int (optional, default None)
+            Bond order.
+        """
+        self._add_TopologyObjects('angles', values, types=types,
+                                 guessed=guessed, order=order)
+    
+    def add_Dihedrals(self, values, types=None, guessed=False, order=None):
+        """Add new Dihedrals to this Universe.
+
+        Parameters
+        ----------
+        values : iterable of tuples, AtomGroups, or Dihedrals
+            An iterable of: tuples of atom indices, or AtomGroups, 
+            or Dihedrals. If every value is a Dihedral, all 
+            keywords are ignored.
+        types : types (optional, default None)
+            Type of dihedral, or atom types
+        guessed : bool (optional, default False)
+            Whether these dihedrals have been guessed.
+        order : int (optional, default None)
+            Bond order.
+        """
+        self._add_TopologyObjects('dihedrals', values, types=types,
+                                 guessed=guessed, order=order)
+    
+    def add_Impropers(self, values, types=None, guessed=False, order=None):
+        """Add new Impropers to this Universe.
+
+        Parameters
+        ----------
+        values : iterable of tuples, AtomGroups, or Impropers
+            An iterable of: tuples of atom indices, or AtomGroups, 
+            or Impropers. If every value is an Improper, all 
+            keywords are ignored.
+        types : types (optional, default None)
+            Type of improper dihedrals, or atom types
+        guessed : bool (optional, default False)
+            Whether these improper dihedrals have been guessed.
+        order : int (optional, default None)
+            Bond order.
+        """
+        self._add_TopologyObjects('impropers', values, types=types,
+                                 guessed=guessed, order=order)
+    
+    
+    
+
+    def _delete_TopologyObjects(self, object_type, values):
+        """Delete TopologyObjects from this Universe
+
+        Parameters
+        ----------
+        object_type : {'bonds', 'angles', 'dihedrals', 'impropers'}
+            The type of TopologyObject to add.
+        values : iterable of tuples, AtomGroups, or TopologyObjects
+            An iterable of: tuples of atom indices, or AtomGroups, 
+            or TopologyObjects.
+        """
+        values = [x.indices if isinstance(x, (AtomGroup, TopologyObject)) 
+                  else x for x in values]
+
+        try:
+            attr = getattr(self._topology, object_type)
+        except AttributeError:
+            raise ValueError('There are no {} to delete'.format(object_type))
+        
+        attr.delete_bonds(values)
+
+    def delete_Bonds(self, values):
+        """Delete Bonds from this Universe and recreate fragments.
```

Comment:
I think it's both irrelevant and confusing for the user that fragments are cleared.